lightning/src/ln/offers_tests.rs

```diff
@@ -727,6 +727,82 @@ fn creates_and_pays_for_offer_using_one_hop_blinded_path() {
 	expect_recent_payment!(bob, RecentPaymentDetails::Fulfilled, payment_id);
 }
 
+/// Verifies that the Invoice Request Message can be retried if it fails to reach the
+/// payee on the first attempt.
+#[test]
+fn creates_and_pays_for_offer_using_one_hop_blinded_path_with_retry() {
```

Comment:
Could you add one for refunds, too?

Reply:
Thanks a lot for the suggestion!

The retry mechanism we've implemented in this PR relies on having the corresponding InvoiceRequest for outbound payments. Since refunds don't have an InvoiceRequest, retrying for them might be a bit tricky.

If you have any ideas on how we can tackle retrying for the Refund case, please do let me know.
Thanks again for all your guidance!

Reply:
No worries at all, Jeff!
Thanks a lot for all your feedback and suggestions! 🌟